util/test/tests/Vulkan/Vk_Pixel_History.py

```diff
@@ -0,0 +1,130 @@
+import renderdoc as rd
+import rdtest
+
+def value_selector(x): return x.floatValue
+def passed(x): return x.Passed()
+def event_id(x): return x.eventId
+def culled(x): return x.backfaceCulled
+def depth_test_failed(x): return x.depthTestFailed
+def stencil_test_failed(x): return x.stencilTestFailed
+def shader_discarded(x): return x.shaderDiscarded
+def shader_out_col(x): return value_selector(x.shaderOut.col)
+def post_mod_col(x): return value_selector(x.postMod.col)
+def primitive_id(x): return x.primitiveID
+
+class VK_Pixel_History(rdtest.TestCase):
+    demos_test_name = 'VK_Pixel_History_Test'
+    demos_frame_cap = 5
+
+    # Need to enable Vulkan Pixel History before opening the capture
+    def run(self):
+        obj: rd.SDObject = rd.SetConfigSetting("Vulkan.PixelHistory")
+        if obj is not None:
+            obj.data.basic.b = True
+        super().run()
+
+    def check_capture(self):
+        apiprops: rd.APIProperties = self.controller.GetAPIProperties()
+
+        if not apiprops.pixelHistory:
+            rdtest.log.print("Vulkan pixel history not tested")
+            return
+
+        test_marker: rd.DrawcallDescription = self.find_draw("Test")
+        self.controller.SetFrameEvent(test_marker.next.eventId, True)
+
+        pipe: rd.PipeState = self.controller.GetPipelineState()
+
+        rt: rd.BoundResource = pipe.GetOutputTargets()[0]
+
+        vp: rd.Viewport = pipe.GetViewport(0)
+
+        tex = rt.resourceId
+        tex_details = self.get_texture(tex)
+
+        sub = rd.Subresource()
+        if tex_details.arraysize > 1:
+            sub.slice = rt.firstSlice
+        if tex_details.mips > 1:
+            sub.mip = rt.firstMip
+
+        # For pixel 190, 149 inside the red triangle
+        x, y = 190, 149
+        rdtest.log.print("Testing pixel {}, {}".format(x, y))
+        modifs: List[rd.PixelModification] = self.controller.PixelHistory(tex, x, y, sub, rt.typeCast)
+        events = [
+            [[event_id, 8], [passed, True]],
+            [[event_id, 12], [passed, True]],
+            [[event_id, 14], [depth_test_failed, True], [post_mod_col, (1.0, 0.0, 0.0, 1.0)]],
+            [[event_id, 19], [stencil_test_failed, True]],
```

Comment:
I've found in the past that hardcoding event IDs in tests gets annoying quickly since if you e.g. add a new state setting command or if the underlying event ID calculation changes then they may all need to be updated. I think it would be better here to insert a marker with a name before each possible modifying event, and then here instead of checking `event_id` directly against a known value, check the name of the drawcall at `event_id-1`.

Reply:
That's a good idea. Done.